### PR TITLE
Include a new optional flag in public-ids that helps generate the sns neuron id

### PR DESCRIPTION
**Pull request by DanielThurau**
# Description

In sns-quill, the public-ids command had an optional memo flag that made calculating the sns neuron id trivial. In the consolidation of quill and sns-quill, this optional flag was not included. The NNS team would like to use this feature in our automated testing and is the only reason we still depend on the sns-quill binary.

# How Has This Been Tested?

Enhanced the existing `tests/commands/public-ids.sh` test.

# Checklist:

- [x] I have made corresponding changes to the documentation in src/docs.

### CODE REVIEW

File: src/commands/public.rs [hunk 1]
```diff
@@ -35,21 +50,27 @@ pub fn exec(auth: &AuthInfo, opts: PublicOpts) -> AnyhowResult {
...
         }
+    };
+
+    match opts.memo {
+        Some(memo) => Ok((
+            principal_id,
+            get_account_id(principal_id)?,
+            Some(get_neuron_id(principal_id, memo)),
+        )),
+        None => Ok((principal_id, get_account_id(principal_id)?, None)),
```
Comment (ericswanson-dfinity) [inline src/commands/public.rs:73]: DRY
```suggestion
    Ok((
        principal_id,
        get_account_id(principal_id, memo)?,
        opts.memo.map(|memo| get_neuron_id(principal_id, memo))
    ))
```

**Comment by adamspofford-dfinity** [comment]
This should probably be a new `quill sns` subcommand instead of an extension of `quill public-ids`.

**Comment by DanielThurau** [comment]
> This should probably be a new quill sns subcommand instead of an extension of quill public-ids.

What do you recommend as the new sns subcommand? Does `quill sns public-ids` sound good to you?

**Comment by adamspofford-dfinity** [comment]
I should think all that's needed is `quill sns neuron-id`, as the rest of the functionality would be covered by `quill public-ids`. Similar to `quill ckbtc deposit-address`.

File: src/commands/sns/neuron_id.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    let neuron_id = NeuronId::from(ledger::compute_neuron_staking_subaccount_bytes(
+        PrincipalId::from(principal_id),
+        opts.memo,
+    ));
+
+    println!("SNS Neuron Id: {}", neuron_id);
+
+    Ok(())
+}
```
Comment (DanielThurau) [inline src/commands/sns/neuron_id.rs:43]:
```suggestion
}

```

File: src/commands/public.rs (hunk 1 above)
Comment (DanielThurau) [reply src/commands/public.rs]: Changed in the next revision.

File: src/commands/sns/neuron_id.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    /// Memo used when calculating the SNS Neuron Id.
+    #[clap(long)]
+    memo: u64,
+}
+
+/// Prints the SNS Neuron Id.
+pub fn exec(auth: &AuthInfo, opts: NeuronIdOpts) -> AnyhowResult {
+    let principal_id = match &opts.principal_id {
+        Some(principal_id) => Principal::from_text(principal_id)?,
```
Comment (adamspofford-dfinity) [inline src/commands/sns/neuron_id.rs:23]: This validation code can be replaced with adding `required_unless_present = "auth"` to `principal_id`.

Comment (DanielThurau) [reply src/commands/sns/neuron_id.rs]: Done.

**Review by adamspofford-dfinity — APPROVED** [review]

**Pull request merged**